plugin/storage/cassandra/factory.go

```diff
@@ -131,6 +137,262 @@ func (f *Factory) configureFromOptions(o *Options) {
 	}
 }
 
+const (
+	MODE               = `MODE`
+	DATACENTER         = `DATACENTER`
+	TRACE_TTL          = `TRACE_TTL`
+	DEPENDENCIES_TTL   = `DEPENDENCIES_TTL`
+	KEYSPACE           = `KEYSPACE`
+	REPLICATION_FACTOR = `REPLICATION_FACTOR`
+	VERSION            = `VERSION`
+	COMPACTION_WINDOW  = `COMPACTION_WINDOW`
+)
+
+// Parameters required for initilizing the db
+type StorageConfigParams struct {
+	mode                   string
+	datacenter             string
+	trace_ttl              int
+	dependencies_ttl       int
+	keyspace               string
+	replication_factor     int
+	replication            string
+	cas_version            int
+	compaction_window_size int
+	compaction_window_unit string
+}
+
+func constructStorageConfigParams() (*StorageConfigParams, error) {
+	var err error
+
+	datacenter := os.Getenv(DATACENTER)
+	replication_factor_string := os.Getenv(REPLICATION_FACTOR)
+
+	var replication_factor, compaction_window_size int
+	var replication, compaction_window_unit string
+
+	mode := os.Getenv(MODE)
```

Comment:
do we actually need the "mode"?

Reply:
In v1 I suppose the mode env variable is used for simple testing scenarios, that's why I added it in this as well. Do you suggest removing it?

Reply:
yes, if we can

Reply:
Removed it. Using `applyDefaults` for apply defaults in case value is not present.